src/components/icon/icon.tsx

```diff
@@ -0,0 +1,76 @@
+import IconSet, { IconType } from './icon-set';
+import React from 'react';
+import styled, { css } from 'styled-components';
+
+export enum IconSize {
+  NORMAL = 'normal', // 24px
+  INHERIT = 'inherit', // width and height equals 150% of the text size of the container
+}
+
+const DEFAULT_ICON_SIZE = IconSize.NORMAL;
+
+const ICON_SIZE_MAP = {
+  [IconSize.NORMAL]: {
+    width: '24px',
+    height: '24px',
+  },
+  [IconSize.INHERIT]: {
+    width: '1.5em',
+    height: '1.5em',
+  },
+};
+
+interface IconContainerProps {
+  size: IconSize;
+}
+
+interface IconProps {
+  /**
+   * Pick an icon by name.
+   */
+  show: IconType /**
+   /**
+   * Choose the size. 
+   */;
+  size?: IconSize;
+  /**
+   * Let's support styled components and friends by accepting this property
+   */
+  className?: string;
+}
+
+const transformSizeToDimension = (size: IconSize) => {
+  if (!Object.keys(ICON_SIZE_MAP).includes(size)) {
```

Comment:
Do we really need this `if`? Through the power of the typescript compiler I feel like we should be perfectly safe here...

Reply:
I had an exception while working on it that's when I did that but actually I should have though like you and ask why is TS not covering my ass. That's the reason:

```
const ICON_SIZE_MAP = {
  [IconSize.NORMAL]: {
    width: '24px',
    height: '24px',
  },
```

This object is not typed even though the only keys are enums. We need to do it like that for enums:

```
type IconSizeMap =  {[key in IconSize]: Dimensions};

const ICON_SIZE_MAP: IconSizeMap = {
```

I have inserted this to make sure values are correct. Thanks.